WalletWasabi.Fluent/ViewModels/Dialogs/CancelTransactionDialogViewModel.cs

```diff
@@ -0,0 +1,81 @@
+using System.Linq;
+using System.Reactive;
+using System.Threading.Tasks;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.TransactionBuilding;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Dialogs;
+
+[NavigationMetaData(Title = "Cancel Transaction")]
+public partial class CancelTransactionDialogViewModel : DialogViewModelBase<Unit>
+{
+	private readonly Wallet _wallet;
+
+	private CancelTransactionDialogViewModel(Wallet wallet, SmartTransaction transactionToCancel, BuildTransactionResult cancellingTransaction)
+	{
+		_wallet = wallet;
+		SetupCancel(enableCancel: true, enableCancelOnEscape: true, enableCancelOnPressed: true);
+
+		var originalFee = transactionToCancel.WalletInputs.Sum(x => x.Amount) - transactionToCancel.OutputValues.Sum(x => x);
+		var cancelFee = cancellingTransaction.Fee;
+		FeeDifference = cancelFee - originalFee;
+		TotalFee = cancelFee;
+		FeeDifferenceUsd = FeeDifference.ToDecimal(MoneyUnit.BTC) * wallet.Synchronizer.UsdExchangeRate;
+		TotalFeeUsd = TotalFee.ToDecimal(MoneyUnit.BTC) * wallet.Synchronizer.UsdExchangeRate;
+
+		EnableBack = false;
+		NextCommand = ReactiveCommand.CreateFromTask(() => OnCancelTransactionAsync(cancellingTransaction));
+	}
+
+	public decimal TotalFeeUsd { get; }
+
+	public Money TotalFee { get; set; }
+
+	public decimal FeeDifferenceUsd { get; }
+
+	public Money FeeDifference { get; }
+
+	protected override void OnDialogClosed()
+	{
+	}
+
+	private async Task OnCancelTransactionAsync(BuildTransactionResult cancellingTransaction)
+	{
+		IsBusy = true;
+
+		try
+		{
+			var isAuthorized = await AuthorizeForPasswordAsync();
+			if (isAuthorized)
+			{
+				await Services.TransactionBroadcaster.SendTransactionAsync(cancellingTransaction.Transaction);
```

Comment:
@nopara73 
After broadcasting a transaction you have to manually update the labels by calling
```c#
_wallet.UpdateUsedHdPubKeysLabels(cancellingTransaction.HdPubKeysWithNewLabels);
```
([more info](https://github.com/zkSNACKs/WalletWasabi/pull/10397))

Should the same be done in the case of cancel? (I am not sure)

Reply:
Or speed up?

Reply:
done